[Bug] Error message when first import

### Bug Description

Hello,

I just installed Gramps Web

If I create manually an individual, no problem but when I want to import a ged file, I have got errors. Same thing with gramps web sync

### Gramps Web Version Information



grampsweb_celery  | [2026-07-30 08:39:10,728: INFO/MainProcess] Task gramps_webapi.api.tasks.import_file[eef0ac82-6234-430a-bf41-25c06f7e4942] received
grampsweb_celery  | [2026-07-30 08:39:11,118: INFO/ForkPoolWorker-1] Task gramps_webapi.api.tasks.import_file[eef0ac82-6234-430a-bf41-25c06f7e4942] ignored
grampsweb-1       | [2026-07-30 08:39:11 +0000] [15] [ERROR] Error handling request GET /api/tasks/eef0ac82-6234-430a-bf41-25c06f7e4942
grampsweb-1       | Traceback (most recent call last):
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 453, in exception_to_python
grampsweb-1       |     exc_type = exc['exc_type']
grampsweb-1       |                ~~~^^^^^^^^^^^^
grampsweb-1       | KeyError: 'exc_type'
grampsweb-1       |
grampsweb-1       | The above exception was the direct cause of the following exception:
grampsweb-1       |
grampsweb-1       | Traceback (most recent call last):
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 141, in handle
grampsweb-1       |     self.handle_request(listener, req, client, addr)
grampsweb-1       |     ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 184, in handle_request
grampsweb-1       |     respiter = self.wsgi(environ, resp.start_response)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1536, in __call__
grampsweb-1       |     return self.wsgi_app(environ, start_response)
grampsweb-1       |            ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1514, in wsgi_app
grampsweb-1       |     response = self.handle_exception(e)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
grampsweb-1       |     response = self.full_dispatch_request()
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
grampsweb-1       |     rv = self.handle_user_exception(e)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
grampsweb-1       |     rv = self.dispatch_request()
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
grampsweb-1       |     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/auth.py", line 44, in wrapper
grampsweb-1       |     return func(*args, **kwargs)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/views.py", line 110, in view
grampsweb-1       |     return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/__init__.py", line 51, in dispatch_request
grampsweb-1       |     return current_app.ensure_sync(meth)(*args, **kwargs)
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask_smorest/response.py", line 89, in wrapper
grampsweb-1       |     current_app.ensure_sync(func)(*args, **kwargs)
grampsweb-1       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/tasks.py", line 159, in get
grampsweb-1       |     "state": task.state,
grampsweb-1       |              ^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/result.py", line 505, in state
grampsweb-1       |     return self._get_task_meta()['status']
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/result.py", line 444, in _get_task_meta
grampsweb-1       |     return self._maybe_set_cache(self.backend.get_task_meta(self.id))
grampsweb-1       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 717, in get_task_meta
grampsweb-1       |     meta = self._get_task_meta_for(task_id)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 1115, in _get_task_meta_for
grampsweb-1       |     return self.decode_result(meta)
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 524, in decode_result
grampsweb-1       |     return self.meta_from_decoded(self.decode(payload))
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 520, in meta_from_decoded
grampsweb-1       |     meta['result'] = self.exception_to_python(meta['result'])
grampsweb-1       |                      ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 455, in exception_to_python
grampsweb-1       |     raise ValueError("Exception information must include "
grampsweb-1       |                      "the exception type") from e
grampsweb-1       | ValueError: Exception information must include the exception type
grampsweb-1       | [2026-07-30 08:39:13 +0000] [15] [ERROR] Error handling request GET /api/tasks/eef0ac82-6234-430a-bf41-25c06f7e4942
grampsweb-1       | Traceback (most recent call last):
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 453, in exception_to_python
grampsweb-1       |     exc_type = exc['exc_type']
grampsweb-1       |                ~~~^^^^^^^^^^^^
grampsweb-1       | KeyError: 'exc_type'
grampsweb-1       |
grampsweb-1       | The above exception was the direct cause of the following exception:
grampsweb-1       |
grampsweb-1       | Traceback (most recent call last):
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 141, in handle
grampsweb-1       |     self.handle_request(listener, req, client, addr)
grampsweb-1       |     ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 184, in handle_request
grampsweb-1       |     respiter = self.wsgi(environ, resp.start_response)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1536, in __call__
grampsweb-1       |     return self.wsgi_app(environ, start_response)
grampsweb-1       |            ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1514, in wsgi_app
grampsweb-1       |     response = self.handle_exception(e)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
grampsweb-1       |     response = self.full_dispatch_request()
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
grampsweb-1       |     rv = self.handle_user_exception(e)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
grampsweb-1       |     rv = self.dispatch_request()
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
grampsweb-1       |     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/auth.py", line 44, in wrapper
grampsweb-1       |     return func(*args, **kwargs)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/views.py", line 110, in view
grampsweb-1       |     return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/__init__.py", line 51, in dispatch_request
grampsweb-1       |     return current_app.ensure_sync(meth)(*args, **kwargs)
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask_smorest/response.py", line 89, in wrapper
grampsweb-1       |     current_app.ensure_sync(func)(*args, **kwargs)
grampsweb-1       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/tasks.py", line 159, in get
grampsweb-1       |     "state": task.state,
grampsweb-1       |              ^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/result.py", line 505, in state
grampsweb-1       |     return self._get_task_meta()['status']
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/result.py", line 444, in _get_task_meta
grampsweb-1       |     return self._maybe_set_cache(self.backend.get_task_meta(self.id))
grampsweb-1       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 717, in get_task_meta
grampsweb-1       |     meta = self._get_task_meta_for(task_id)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 1115, in _get_task_meta_for
grampsweb-1       |     return self.decode_result(meta)
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 524, in decode_result
grampsweb-1       |     return self.meta_from_decoded(self.decode(payload))
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 520, in meta_from_decoded
grampsweb-1       |     meta['result'] = self.exception_to_python(meta['result'])
grampsweb-1       |                      ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 455, in exception_to_python
grampsweb-1       |     raise ValueError("Exception information must include "
grampsweb-1       |                      "the exception type") from e
grampsweb-1       | ValueError: Exception information must include the exception type
grampsweb-1       | [2026-07-30 08:39:14 +0000] [13] [ERROR] Error handling request GET /api/tasks/eef0ac82-6234-430a-bf41-25c06f7e4942
grampsweb-1       | Traceback (most recent call last):
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 453, in exception_to_python
grampsweb-1       |     exc_type = exc['exc_type']
grampsweb-1       |                ~~~^^^^^^^^^^^^
grampsweb-1       | KeyError: 'exc_type'
grampsweb-1       |
grampsweb-1       | The above exception was the direct cause of the following exception:
grampsweb-1       |
grampsweb-1       | Traceback (most recent call last):
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 141, in handle
grampsweb-1       |     self.handle_request(listener, req, client, addr)
grampsweb-1       |     ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 184, in handle_request
grampsweb-1       |     respiter = self.wsgi(environ, resp.start_response)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1536, in __call__
grampsweb-1       |     return self.wsgi_app(environ, start_response)
grampsweb-1       |            ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1514, in wsgi_app
grampsweb-1       |     response = self.handle_exception(e)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
grampsweb-1       |     response = self.full_dispatch_request()
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
grampsweb-1       |     rv = self.handle_user_exception(e)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
grampsweb-1       |     rv = self.dispatch_request()
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
grampsweb-1       |     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/auth.py", line 44, in wrapper
grampsweb-1       |     return func(*args, **kwargs)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/views.py", line 110, in view
grampsweb-1       |     return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/__init__.py", line 51, in dispatch_request
grampsweb-1       |     return current_app.ensure_sync(meth)(*args, **kwargs)
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask_smorest/response.py", line 89, in wrapper
grampsweb-1       |     current_app.ensure_sync(func)(*args, **kwargs)
grampsweb-1       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/tasks.py", line 159, in get
grampsweb-1       |     "state": task.state,
grampsweb-1       |              ^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/result.py", line 505, in state
grampsweb-1       |     return self._get_task_meta()['status']
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/result.py", line 444, in _get_task_meta
grampsweb-1       |     return self._maybe_set_cache(self.backend.get_task_meta(self.id))
grampsweb-1       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 717, in get_task_meta
grampsweb-1       |     meta = self._get_task_meta_for(task_id)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 1115, in _get_task_meta_for
grampsweb-1       |     return self.decode_result(meta)
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 524, in decode_result
grampsweb-1       |     return self.meta_from_decoded(self.decode(payload))
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 520, in meta_from_decoded
grampsweb-1       |     meta['result'] = self.exception_to_python(meta['result'])
grampsweb-1       |                      ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 455, in exception_to_python
grampsweb-1       |     raise ValueError("Exception information must include "
grampsweb-1       |                      "the exception type") from e
grampsweb-1       | ValueError: Exception information must include the exception type
grampsweb-1       | [2026-07-30 08:39:15 +0000] [16] [ERROR] Error handling request GET /api/tasks/eef0ac82-6234-430a-bf41-25c06f7e4942
grampsweb-1       | Traceback (most recent call last):
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 453, in exception_to_python
grampsweb-1       |     exc_type = exc['exc_type']
grampsweb-1       |                ~~~^^^^^^^^^^^^
grampsweb-1       | KeyError: 'exc_type'
grampsweb-1       |
grampsweb-1       | The above exception was the direct cause of the following exception:
grampsweb-1       |
grampsweb-1       | Traceback (most recent call last):
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 141, in handle
grampsweb-1       |     self.handle_request(listener, req, client, addr)
grampsweb-1       |     ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 184, in handle_request
grampsweb-1       |     respiter = self.wsgi(environ, resp.start_response)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1536, in __call__
grampsweb-1       |     return self.wsgi_app(environ, start_response)
grampsweb-1       |            ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1514, in wsgi_app
grampsweb-1       |     response = self.handle_exception(e)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
grampsweb-1       |     response = self.full_dispatch_request()
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
grampsweb-1       |     rv = self.handle_user_exception(e)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
grampsweb-1       |     rv = self.dispatch_request()
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
grampsweb-1       |     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/auth.py", line 44, in wrapper
grampsweb-1       |     return func(*args, **kwargs)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask/views.py", line 110, in view
grampsweb-1       |     return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/__init__.py", line 51, in dispatch_request
grampsweb-1       |     return current_app.ensure_sync(meth)(*args, **kwargs)
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/flask_smorest/response.py", line 89, in wrapper
grampsweb-1       |     current_app.ensure_sync(func)(*args, **kwargs)
grampsweb-1       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/tasks.py", line 159, in get
grampsweb-1       |     "state": task.state,
grampsweb-1       |              ^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/result.py", line 505, in state
grampsweb-1       |     return self._get_task_meta()['status']
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/result.py", line 444, in _get_task_meta
grampsweb-1       |     return self._maybe_set_cache(self.backend.get_task_meta(self.id))
grampsweb-1       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 717, in get_task_meta
grampsweb-1       |     meta = self._get_task_meta_for(task_id)
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 1115, in _get_task_meta_for
grampsweb-1       |     return self.decode_result(meta)
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 524, in decode_result
grampsweb-1       |     return self.meta_from_decoded(self.decode(payload))
grampsweb-1       |            ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 520, in meta_from_decoded
grampsweb-1       |     meta['result'] = self.exception_to_python(meta['result'])
grampsweb-1       |                      ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
grampsweb-1       |   File "/venv/lib/python3.13/site-packages/celery/backends/base.py", line 455, in exception_to_python
grampsweb-1       |     raise ValueError("Exception information must include "
grampsweb-1       |                      "the exception type") from e
grampsweb-1       | ValueError: Exception information must include the exception type


### Verification

- [x] I have searched for existing issues before creating this one

--- a/gramps_webapi/api/resources/tasks.py
+++ b/gramps_webapi/api/resources/tasks.py
@@ -20,6 +20,7 @@
 """Background task resources."""
 
 from http import HTTPStatus
+from typing import Any, Dict, NamedTuple, Optional
 
 from celery.result import AsyncResult
 from flask import abort
@@ -139,6 +140,41 @@ def _serializable_task_result(obj):
         return str(obj)
 
 
+class TaskMeta(NamedTuple):
+    """State and result of a task as reported by the result backend."""
+
+    state: str
+    result: Any
+
+
+def _task_error_payload(result: Any) -> Optional[Dict]:
+    """Return the TaskError payload of a failed task, if it has one.
+
+    Celery rebuilds TaskError as a synthesised class when the module is not
+    importable, so match on the shape rather than on the type.
+    """
+    if isinstance(result, BaseException) and len(result.args) == 1:
+        payload = result.args[0]
+        if isinstance(payload, dict) and "error" in payload:
+            return payload
+    return None
+
+
+def _task_meta(task: AsyncResult) -> TaskMeta:
+    """Return state and result of a task, unwrapping aborts to their payload.
+
+    Aborts stored by versions predating TaskError cannot be decoded by Celery,
+    which raises ValueError; report those as an opaque failure until they
+    expire from the result backend.
+    """
+    try:
+        state, result = task.state, task.result
+    except ValueError:
+        return TaskMeta("FAILURE", "Task failed with an unreadable result")
+    payload = _task_error_payload(result)
+    return TaskMeta(state, result if payload is None else payload)
+
+
 class TaskResource(ProtectedResource):
     """Resource for a single task."""
 
@@ -155,12 +191,13 @@ class TaskResource(ProtectedResource):
             if row.tree != tree:
                 abort(HTTPStatus.FORBIDDEN)
 
+        meta = _task_meta(task)
         result = {
-            "state": task.state,
-            "result_object": _serializable_task_result(task.result),
-            # kept for backward compatibility
-            "info": _serialize_task_result(task.info),
-            "result": _serialize_task_result(task.result),
+            "state": meta.state,
+            "result_object": _serializable_task_result(meta.result),
+            # kept for backward compatibility (info is an alias of result)
+            "info": _serialize_task_result(meta.result),
+            "result": _serialize_task_result(meta.result),
         }
         if row is not None:
             result["task_id"] = row.task_id
@@ -215,7 +252,7 @@ class TaskListResource(ProtectedResource):
                 "user_id": row.user_id,
                 "user_name": user_name_map.get(row.user_id) if row.user_id else None,
                 "state": (
-                    (AsyncResult(row.task_id).state or "PENDING")
+                    (_task_meta(AsyncResult(row.task_id)).state or "PENDING")
                     if args["include_state"]
                     else None
                 ),

--- a/gramps_webapi/util/celery.py
+++ b/gramps_webapi/util/celery.py
@@ -2,8 +2,15 @@
 
 from celery import Task
 from celery import current_app as current_celery_app
-from celery.exceptions import Ignore
 from werkzeug.exceptions import HTTPException
+
+
+class TaskError(Exception):
+    """Task failure carrying an API-style error payload as its only argument.
+
+    Celery serialises exception args, so the payload survives the round trip
+    through the result backend.
+    """
 
 
 def create_celery(app):
@@ -24,18 +31,14 @@ def create_celery(app):
                 try:
                     return self.run(*args, **kwargs)
                 except HTTPException as exc:
-                    # Utility functions like check_quota_people and run_import
-                    # use abort_with_message (a Flask/HTTP construct) for both
-                    # request handlers and Celery tasks. Preserve the standard
-                    # API error shape here so task consumers do not need
-                    # special-case parsing for background-task failures.
-                    # Ideally these utilities would raise a dedicated
-                    # TaskError instead of an HTTPException.
-                    self.update_state(
-                        state="FAILURE",
-                        meta={"error": {"code": exc.code, "message": exc.description}},
-                    )
-                    raise Ignore()
+                    # Utilities like check_quota_people abort with an
+                    # HTTPException inside tasks too; preserve the API error
+                    # shape. It must be raised, not stored via update_state:
+                    # Celery can only decode serialised exceptions in exception
+                    # states, and chokes on a plain dict when reading it back.
+                    raise TaskError(
+                        {"error": {"code": exc.code, "message": exc.description}}
+                    ) from exc
 
     celery.Task = ContextTask
     return celery

--- a/tests/test_endpoints/test_tasks.py
+++ b/tests/test_endpoints/test_tasks.py
@@ -390,6 +390,27 @@ class TestTaskEndpoints(unittest.TestCase):
         assert rv.status_code == 200
         assert rv.json["user_name"] is None
 
+    def test_get_task_failed_with_abort(self):
+        """A task aborted via abort_with_message reports the API error shape."""
+        from gramps_webapi.util.celery import TaskError
+
+        payload = {"error": {"code": 405, "message": "Not allowed by people quota"}}
+        task_id = str(uuid.uuid4())
+        self._insert_row(task_id, name="import_file", user_id=self.owner_id)
+        task = Mock()
+        task.state = "FAILURE"
+        task.result = TaskError(payload)
+        with patch("gramps_webapi.api.resources.tasks.AsyncResult", return_value=task):
+            rv = self.client.get(f"/api/tasks/{task_id}", headers=self._auth("owner"))
+        assert rv.status_code == 200
+        assert rv.json["state"] == "FAILURE"
+        assert rv.json["result_object"] == payload
+        assert (
+            rv.json["result"]
+            == '{"error":{"code":405,"message":"Not allowed by people quota"}}'
+        )
+        assert rv.json["info"] == rv.json["result"]
+
     def test_list_excludes_expired_tasks(self):
         """Tasks older than the result TTL are excluded from the list."""
         fresh_id = str(uuid.uuid4())

--- a/tests/test_task_error.py
+++ b/tests/test_task_error.py
@@ -1,0 +1,107 @@
+#
+# Gramps Web API - A RESTful API for the Gramps genealogy program
+#
+# Copyright (C) 2026      David Straub
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Tests for reporting aborts that happen inside a background task."""
+
+import unittest
+
+import pytest
+from celery import Celery
+from celery.backends.base import Backend
+
+from gramps_webapi.api.resources.tasks import _task_error_payload, _task_meta
+from gramps_webapi.util.celery import TaskError
+
+PAYLOAD = {"error": {"code": 405, "message": "Not allowed by people quota"}}
+
+
+def _backend(serializer="json"):
+    """Return a result backend not connected to any broker."""
+    return Backend(app=Celery(), serializer=serializer)
+
+
+class TestTaskErrorSerialization(unittest.TestCase):
+    """The error payload must survive a round trip through the result backend."""
+
+    def test_round_trip(self):
+        backend = _backend()
+        stored = backend.decode(
+            backend.encode(backend.prepare_exception(TaskError(PAYLOAD)))
+        )
+        exc = backend.exception_to_python(stored)
+        assert isinstance(exc, TaskError)
+        assert exc.args[0] == PAYLOAD
+
+    def test_round_trip_meta(self):
+        """The full meta dict as stored for a failed task decodes fine."""
+        backend = _backend()
+        meta = {
+            "status": "FAILURE",
+            "result": backend.prepare_exception(TaskError(PAYLOAD)),
+        }
+        decoded = backend.meta_from_decoded(backend.decode(backend.encode(meta)))
+        assert decoded["status"] == "FAILURE"
+        assert _task_error_payload(decoded["result"]) == PAYLOAD
+
+    def test_plain_dict_result_is_unreadable(self):
+        """Storing the payload directly under FAILURE poisons the result.
+
+        This is what update_state(state="FAILURE", meta=payload) used to do and
+        the reason TaskError exists; Celery refuses to decode it.
+        """
+        backend = _backend()
+        with pytest.raises(ValueError):
+            backend.meta_from_decoded({"status": "FAILURE", "result": PAYLOAD})
+
+
+class TestTaskErrorPayload(unittest.TestCase):
+    """Extraction of the API error shape from a task result."""
+
+    def test_task_error(self):
+        assert _task_error_payload(TaskError(PAYLOAD)) == PAYLOAD
+
+    def test_synthesized_exception_class(self):
+        """Celery rebuilds unknown exception types dynamically."""
+        cls = type("TaskError", (Exception,), {})
+        assert _task_error_payload(cls(PAYLOAD)) == PAYLOAD
+
+    def test_other_exception(self):
+        assert _task_error_payload(ValueError("Failed importing gramps file")) is None
+
+    def test_non_exception_result(self):
+        assert _task_error_payload({"people": 3}) is None
+
+
+class _PoisonedAsyncResult:
+    """Result stored in the pre-TaskError shape, which Celery cannot decode."""
+
+    @property
+    def state(self):
+        raise ValueError("Exception information must include the exception type")
+
+    result = state
+
+
+class TestLegacyResultFallback(unittest.TestCase):
+    """Results stored in the pre-TaskError shape must not raise."""
+
+    def test_does_not_raise(self):
+        meta = _task_meta(_PoisonedAsyncResult())
+        assert meta.state == "FAILURE"
+        assert isinstance(meta.result, str)

